bot.ml

```diff
@@ -750,6 +796,24 @@ let callback _conn req body =
       | Error s ->
           Server.respond_string ~status:(Code.status_of_code 400)
             ~body:(f "Error: %s" s) () )
+  | "/coq-bug-minimizer" ->
+      body
+      >>= fun body ->
+      if string_match ~regexp:"\\([^\n]+\\)\n\\([^\r]*\\)" body then
+        let stamp = Str.matched_group 1 body in
+        let message = Str.matched_group 2 body in
+        match Str.split (Str.regexp " ") stamp with
+        | [id; author; repo_name; branch_name] ->
+            GitHub_mutations.post_comment ~token:github_access_token ~id
+              ~message:(f "@%s, %s" author message)
+            >>= fun () ->
+            execute_cmd
+              (f "git push https://%s:%s@github.com/%s.git --delete '%s'"
+                 bot_name github_access_token repo_name branch_name)
+            >>= fun _ -> Server.respond_string ~status:`OK ~body:"" ()
```

Comment:
Since the response doesn't depend on the results on the previous action, only on the fact that the parsing of the body went well, you can send it asynchronously. The git delete and post comment actions can also be run in parallel (cf. the `<&>` operator).